test/blackbox-tests/test-cases/link-deps.t/run.t

```diff
@@ -4,11 +4,8 @@ In particular, these can depend on the result of the compilation (like a .cmo
 file) and be created just before linking.
 
   $ dune build link_deps.exe
-  File "dune", line 1, characters 0-105:
-  1 | (alias
-  2 |   (name message)
-  3 |   (deps .link_deps.eobjs/.byte_objs/link_deps.cmo)
-  4 |   (action (echo "link\n"))
-  5 |   )
+  File "dune", line 8, characters 8-17:
+  8 |   (name link_deps)
```

Comment:
Did this error message improve? The previous one makes more sense to me.

Reply:
Yeah, indeed. I need to investigate this

Reply:
Ok, I understand what's going on. At the point where we raise this error, we don't have a location. To recover the location, we inspect the memo stack, observe a call to `execute_rule rule` and take the `rule.loc`. However, we don't memoize calls to `execute_action` given that these are made on temporary rules built on the fly. As a result, the location of the `alias` stanza doesn't appear in the memo stack and the first one to appear is the location of the rule that depends on this alias.

It feels like we want to be able to "inject" the location in the memo stack somehow.

Reply:
I have a fix for it, but it's bigger change, so I'll submit as a separate PR.